src/dmd/lexer.d

```diff
@@ -1955,6 +1969,14 @@ class Lexer : ErrorHandler
             error("integer overflow");
             err = true;
         }
+        if ((base == 2 && !anyBinaryDigitsUS) ||
+            (base == 16 && !anyHexDigitsNoSingleUS))
+        {
+            auto tmp = new char[p-start+1];
+            tmp[0 .. p-start] = start[0 .. p-start];
+            tmp[$-1] = cast(char) null;
+            deprecation("%s isn't a valid integer literal", tmp.ptr);
```

Comment:
Code that is quoted in messages should be surrounded by tick marks.  In other words...

```
`%s` isn't a valid integer literal
```